### PR TITLE
Add a program label to shader compile errors

### PR DESCRIPTION
**Pull request by mbouron-gpsw**

### CODE REVIEW

File: libnodegl/pass.h [hunk 1]
```diff
@@ -31,6 +31,7 @@ struct ngl_ctx;
 
 struct pass_params {
     const char *label;
+    const char* program_label;
```
Comment (cboesch-gpsw) [inline libnodegl/pass.h:34]: misplaced `*`

File: libnodegl/backends/gl/program_gl.c [hunk 2]
```diff
@@ -289,7 +289,8 @@ int ngli_program_gl_init(struct program *s, const struct program_params *params)
         if (ret < 0) {
             char *s_with_numbers = ngli_numbered_lines(shaders[i].src);
             if (s_with_numbers) {
-                LOG(ERROR, "failed to compile:\n%s", s_with_numbers);
+                LOG(ERROR, "failed to compile shader (label=%s):\n%s",
+                    params->label ? params->label : "undefined", s_with_numbers);
```
Comment (cboesch-gpsw) [inline libnodegl/backends/gl/program_gl.c:294]: nitpick, I'd prefer:
```c
                LOG(ERROR, "failed to compile shader \"%s\":\n%s",
                    params->label ? params->label : "", s_with_numbers);
```

Can we also add a commit to assert that it is always set after every code unit has been updated so we can remove the conditional printing?

File: libnodegl/backends/gl/program_gl.c [hunk 3]
```diff
@@ -271,7 +271,8 @@ int ngli_program_gl_init(struct program *s, const struct program_params *params)
     struct gpu_ctx_gl *gpu_ctx_gl = (struct gpu_ctx_gl *)s->gpu_ctx;
     struct glcontext *gl = gpu_ctx_gl->glcontext;
 
-    if (params->compute && (gl->features & NGLI_FEATURE_GL_COMPUTE_SHADER_ALL) != NGLI_FEATURE_GL_COMPUTE_SHADER_ALL) {
+    const uint64_t features = NGLI_FEATURE_GL_COMPUTE_SHADER_ALL;
+    if (params->compute && (gl->features & features) != features) {
```
Comment (cboesch-gpsw) [inline libnodegl/backends/gl/program_gl.c:275]: Alternatively you could expose `libnodegl/api.c:#define ALL_FEATURES(features, mask) ((features & (mask)) == mask)` internally and use that.

Not a blocking comment.

File: libnodegl/hud.c [hunk 4]
```diff
@@ -1263,6 +1263,7 @@ int ngli_hud_init(struct hud *s)
     graphicstate.blend_dst_factor_a = NGLI_BLEND_FACTOR_ONE;
 
     const struct pgcraft_params crafter_params = {
+        .program_label    = "hud",
```
Comment (cboesch-gpsw) [inline libnodegl/hud.c:1266]: Shouldn't we make a distinction between user label and internal labels? Maybe something like "internal/hud"", "nodegl/hud", "ngl-hud", ...

**Review by cboesch-gpsw — CHANGES_REQUESTED** [review]
Small nitpicks, LGTM otherwise

File: libnodegl/hud.c (hunk 4 above)
Comment (mbouron-gpsw) [reply libnodegl/hud.c:1266]: Picked `nodegl/`.

File: libnodegl/pass.h (hunk 1 above)
Comment (mbouron-gpsw) [reply libnodegl/pass.h]: Fixed.

File: libnodegl/backends/gl/program_gl.c (hunk 3 above)
Comment (mbouron-gpsw) [reply libnodegl/backends/gl/program_gl.c]: Will do something like that later on for the whole code base since we use this pattern in other places.

**Review by cboesch-gpsw — APPROVED** [review]

File: libnodegl/backends/gl/program_gl.c (hunk 2 above)
Comment (mbouron-gpsw) [reply libnodegl/backends/gl/program_gl.c]: I prefer to keep the label optional for now.

**Pull request merged**